### DISCUSSION TITLE
Daily Story

### DISCUSSION THREAD

**Discussion by kentcdodds** [johnlindquist/kit#357] [Share]
Wrote a new script where I can quickly write little stories of what's going on in my life. It's kind of like a little journal. I set the `DAILY_STORY_DIRECTORY` to a folder in dropbox.

The only thing I'm unsure of right now is I want to use `textarea` but I don't know how to actually submit what's been typed. Hitting "enter" just (correctly) inserts a newline. I tried ⌘+enter and that didn't work. Anyone know how to submit a textarea?

[Install daily-story](https://scriptkit.com/api/new?name=daily-story&url=https://gist.githubusercontent.com/kentcdodds/1a5d2b4eabd90ae6b45734ddc2a040dc/raw/d59daeb69d8ef54d54fc4823c0b7f2c147d6932e/daily-story.js)

```js
// Menu: Daily Story
// Description: Write a quick story
// Author: Kent C. Dodds
// Shortcut: command option control o
// Twitter: @kentcdodds

const dateFns = await npm('date-fns')
const filenamify = await npm('filenamify')
const prettier = await npm('prettier')

const storyDir = await env(
  'DAILY_STORY_DIRECTORY',
  `Where do you want daily stories to be saved?`,
)

const story = await arg({placeholder: 'Write your story here'})

const today = dateFns.format(new Date(), 'yyyy-MM-dd')
const date = await arg({
  input: today,
  hint: 'When did this happen?',
})
const title = await arg({
  placeholder: 'What do you want to call this story?',
  hint: 'Title',
})

const md = `---
title: ${title}
date: ${date}
written: ${today}
---

${story}
`

// prettify the markdown
const prettyMd = await prettier.format(md, {
  parser: 'markdown',
  arrowParens: 'avoid',
  bracketSpacing: false,
  embeddedLanguageFormatting: 'auto',
  htmlWhitespaceSensitivity: 'css',
  insertPragma: false,
  jsxBracketSameLine: false,
  jsxSingleQuote: false,
  printWidth: 80,
  proseWrap: 'always',
  quoteProps: 'as-needed',
  requirePragma: false,
  semi: false,
  singleQuote: true,
  tabWidth: 2,
  trailingComma: 'all',
  useTabs: false,
  vueIndentScriptAndStyle: false,
})

const filename = filenamify(
  `${date}-${title.toLowerCase().replace(/ /g, '-')}.md`,
  {replacement: '-'},
)
await writeFile(path.join(storyDir, filename), prettyMd)

```

**Comment by johnlindquist** [comment]
Love it! 😍

I use similar scripts for my journal, gratitude, and Script Kit notes.

> "Anyone know how to submit a textarea?"

cmd + s submits
cmd + w cancels

in both the `textarea` and `editor` components. (reminder that `editor` will be a paid upgrade in the future)

**Reply by kentcdodds** [reply]
Sweet, thanks!

I tried `editor` and I'm not sure what the benefit of `editor` is over `textarea` 🤔

**Reply by johnlindquist** [reply]
1. Find/replace

Just use cmd+f

2. Other shortcuts:
Essentially the same as this: https://docs.monaca.io/en/products_guide/monaca_ide/editor/

3. Monaco editor options:

```js
/** @type typof import("kit") */
// ☝️ If Kit's hints/types aren't showing, this comment can kick VS Code into including them

let value = `Wrote a new script where I can quickly write little stories of what's going on in my life. It's kind of like a little journal. I set the DAILY_STORY_DIRECTORY to a folder in dropbox.

The only thing I'm unsure of right now is I want to use textarea but I don't know how to actually submit what's been typed. Hitting "enter" just (correctly) inserts a newline. I tried ⌘+enter and that didn't work. Anyone know how to submit a textarea?`

await editor({
  value,
  language: "markdown",
  fontFamily: "Dank Mono",
  fontSize: "20px",
  wordBasedSuggestions: true,
})
```

**Reply by johnlindquist** [reply]
Oh, and I'll eventually add diff'ing and bunch more stuff 😉

**Reply by joelgrimberg** [reply]
at the moment the Monaco editor options do not work yet with predefined content:
```javascript
let newContent = await editor(contents, {
    language: 'markdown',
    fontFamily: 'Dank Mono',
    fontSize: '40px',
    wordBasedSuggestions: true,
  })
  ```

**Reply by johnlindquist** [reply]
Ah, the api is unclear. The first arg is a string or an options object. The `value` of the options object would be the `content`:

[Open editor-options-with-content in Script Kit](https://scriptkit.com/api/new?name=editor-options-with-content&url=https://gist.githubusercontent.com/johnlindquist/4445367f731a4946bce17dccf1737b74/raw/34517f3e2daeec16eb378c49a0bac5a1ed6a2c12/editor-options-with-content.ts")

```js
// Name: Editor Options with Content

import "@johnlindquist/kit"

let content = `# Hello world`

let newContent = await editor({
    value: content,
    fontFamily: "Dank Mono",
    fontSize: "40px",
    wordBasedSuggestions: true
})

dev(newContent)
```
